msbuild/Xamarin.MacDev.Tasks/Tasks/AnalyzeFileChanges.cs

```diff
@@ -0,0 +1,126 @@
+
+		//In case it's a remote execution, we don't want empty output files to be copied since we need the real files to be copied
+		public bool ShouldCreateOutputFile (ITaskItem item) => false;
+
+		public override bool Execute ()
+		{
+			if (ShouldExecuteRemotely ()) {
+				return new TaskRunner (SessionId, BuildEngine4).RunAsync (this).Result;
+			}
+
+			if (!Directory.Exists (WorkingDirectory)) {
+				Log.LogError ("Unable to analyze file changes in '{0}'. Directory not found.", WorkingDirectory);
+
+				return false;
+			}
+
+			if (!File.Exists (ReportFile!.ItemSpec)) {
+				Log.LogError ("Unable to analyze file changes in '{0}'. The report file '{1}' does not exist", WorkingDirectory, ReportFile.ItemSpec);
+
+				return false;
+			}
+
+			try {
+				var changedFiles = new List<ITaskItem> ();
+				//Gets a dictionary of file names, lengths and MVIDs from the ReportFile
+				IDictionary<string, (long length, Guid mvid)> reportFileList = GetReportFileList ();
+				IEnumerable<string> files = Directory
+					.GetFiles (WorkingDirectory, "*", SearchOption.TopDirectoryOnly)
+					.Where (f => Path.GetExtension (f) == ".dll");
+
+				foreach (string file in files) {
+					//We only want the files that are present in the ReportFile
+					if (!reportFileList.TryGetValue (Path.GetFileName (file), out (long length, Guid mvid) localInfo)) {
+						continue;
+					}
+
+					var fileInfo = new FileInfo (file);
+
+					//If the file lengths differ, it means local and remote versions are different
+					if (fileInfo.Length != localInfo.length) {
+						changedFiles.Add (new TaskItem (file));
+						TryAddPdbFile (file, changedFiles);
+
+						continue;
+					}
+
+					using Stream stream = fileInfo.OpenRead ();
+					using var peReader = new PEReader (stream);
+					MetadataReader metadataReader = peReader.GetMetadataReader ();
+					Guid mvid = metadataReader.GetGuid (metadataReader.GetModuleDefinition ().Mvid);
+
+					//If the MVID from the report file (local MVID) is different than the calculated MVID of the file, it means local and remote versions are different
+					if (mvid != localInfo.mvid) {
+						changedFiles.Add (new TaskItem (file));
+						TryAddPdbFile (file, changedFiles);
+					}
+				}
+
+				ChangedFiles = [.. changedFiles];
+
+				return true;
+			} catch (Exception ex) {
+				Log.LogError ("Unable to analyze file changes in '{0}'. An unexpected error occurred.", WorkingDirectory);
+				Log.LogErrorFromException (ex);
+
+				return false;
+			}
+		}
+
+		IDictionary<string, (long length, Guid mvid)> GetReportFileList ()
+		{
+			var reportFileList = new Dictionary<string, (long length, Guid mvid)> ();
+
+			//Expected format of the report file lines (defined in the CalculateAssembliesReport task): Foo.dll/23189/768C814C-05C3-4563-9B53-35FEF571968E
+			foreach (var line in File.ReadLines (ReportFile!.ItemSpec)) {
+				string [] lineParts = line.Split (["/"], StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Ever so slight performance improvement:

```suggestion
				string [] lineParts = line.Split (['/'], StringSplitOptions.RemoveEmptyEntries);
```